### ISSUE TITLE
Libraries are not being loaded

### ISSUE THREAD

**Issue by eldan88** [kenjis/ci-phpunit-test#113]
This is the main issue I am running into..

Here is my controller class

```

class Welcome extends CI_Controller {

    public function __construct()
    {

        parent::__construct();
        $this->load->library(["form_validation","Mcrypt"]);
        $this->load->helper(["html"]);

    }


    public function testCode(){

        return "hello world";

    }

}

```

Here is my test class

```

class Welcome_test extends TestCase
{

    public function test_index()
    {
        $this->request("GET", "Welcome");
        $obj = new Welcome();
        $hi = $obj->testCode();
        $this->assertEquals("hi", $hi);


    }
}

```

When I run the test my libraries won't load. For example the mcrypt library does not load

I keep on getting this issue

```

1) Welcome_test::test_index
CIPHPUnitTestShowErrorException: Unable to load the requested class: Mcrypt

/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/_ci_phpunit_test/replacing/core/Common.php:183
/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/_ci_phpunit_test/replacing/core/Loader.php:1102
/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/_ci_phpunit_test/replacing/core/Loader.php:1097
/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/_ci_phpunit_test/replacing/core/Loader.php:218
/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/_ci_phpunit_test/replacing/core/Loader.php:202
/Users/eldanshkolnikov/Dropbox/web-content/ci/application/tests/controllers/Welcome_test.php:23
```

**Comment by kenjis** [comment]
I can't reproduce your error.
See https://github.com/kenjis/ci-phpunit-test-113

~~~
$ cd application/tests/
$ phpunit
PHPUnit 4.8.24 by Sebastian Bergmann and contributors.

F

Time: 785 ms, Memory: 6.00Mb

There was 1 failure:

1) Welcome_test::test_index
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'hi'
+'hello world'

.../ci-phpunit-test-113/application/tests/controllers/Welcome_test.php:10

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
~~~

**Comment by eldan88** [comment]
Can you please provide your controller code and your test code?

**Comment by kenjis** [comment]
See
* https://github.com/kenjis/ci-phpunit-test-113/blob/master/application/controllers/Welcome.php
* https://github.com/kenjis/ci-phpunit-test-113/blob/master/application/tests/controllers/Welcome_test.php

You can download:
https://github.com/kenjis/ci-phpunit-test-113/archive/master.zip

**Comment by eldan88** [comment]
Oh I see what you did, you have uploaded the whole repo. I will clone it and test it as well. Thanks a lot for your helping me out, I really want to unit test my project. It sucks that codeigniter classes are so tightly coupled. 

I don't know if you heard but codeigniter 4 is going to have dependency injection built into it 
http://forum.codeigniter.com/thread-64552.html

This should make testing a whole lot easier.

**Comment by eldan88** [comment]
I downloaded the code you provided, I cd into the tests folder, and ran the phpunit command and got the following error message

`Your system folder path does not appear to be set correctly. Please open the following file and correct this: Bootstrap.php`

**Comment by kenjis** [comment]
@eldan88 You need to run `composer install` in the project root directory.
If you don't have `composer`, see <https://getcomposer.org/download/>.

**Comment by eldan88** [comment]
Okay. I ran composer install and it seems to be finally working. I now need  try this again on my actual project. 

Can you please whats mikey179, and why is composer being installed again in the vendor folder? Thnaks

**Comment by kenjis** [comment]
`mikey179/vfsStream` is a Composer package which provides a virtual file system.
It is used for CodeIgniter's unit testing. Probably you don't use it.

> why is composer being installed again in the vendor folder? 

What do you mean by the *composer*? The `vendor` folder is the place where Composer install dependency packages.

The `vendor/composer` is the place where Composer puts its autoloader files and installed package information.

**Comment by eldan88** [comment]
I am sorry. I meant to ask why is codeigniter framework being installed again.

**Comment by kenjis** [comment]
No problem.
CodeIginter framework is installed only once.
If you see the repository <https://github.com/kenjis/ci-phpunit-test-113> carefully,
you see there is no the `system` folder.

I create the repository by <https://github.com/kenjis/codeigniter-composer-installer>.

**Comment by kenjis** [comment]
If you still have questions, feel free to create another issue.
